NOTES.md

```diff
@@ -0,0 +1,60 @@
+# Notes from review
+
+* WebKit's documentation and intro on parking lot and its paradigm is quite good
+
+* `cargo test` fails with compilation errors on Linux
+  * had an ancient lock file on my system and `cargo update` was needed
+
+* Various bits and pieces of the library are centered around compatibility with
+  various versions of Rust. This naturally isn't needed when integrating into
+  libstd, but it can often reduce the readability of code and/or make it more
+  difficult to follow. Do we want to have a clear version policy for parking_lot
+  on crates.io which is geared towards simplifying the code?
+
+* Overall quite worried about atomic orderings. Everything should be `SeqCst`
+  until proven otherwise, and even then needs quite a lot of proof (IMO) to not
+  use `SeqCst`. Being fast-and-loose with orderings seems like we're trying to
+  be too clever by half.
+
+* The `ThreadParker` API seems pretty subtle. For example the `UnparkHandle`
+  type can outlive the `ThreadParker` itself but that's ok. This seems currently
+  undocumented?
+
+* Can `parking_lot::Mutex` be used in as many places as the current OS-based
+  mutex? I think so but it's worth considering. For example you may be able to
+  use `std::sync::Mutex` today to protect a global allocator, but with parking
+  lot you won't be able to. Similarly about thread local internals with libstd,
+  but I think it's just internal stuff to libstd.
```

Comment:
You make a good point about the dependency on the global allocator. `park` may need to grow the hash table if there are many parked threads. I'm not really sure if there is a good way to resolve this issue, maybe we could force parking_lot to use the system allocator?

Reply:
I don't think there's really a great answer to this either, I think it's just something we'll have to acknowledge as the libs team and be sure to indicate in release notes and/or changelogs. If it becomes a big issue we can try to use the system allocator or some other mechanism, but in general we don't have a great fix for this.

Reply:
On Redox, for example, I believe that the Rust global allocator *is* the system allocator!  So `parking_lot` won’t work in that case.  Furthermore, some code (such as OS kernels, filesystem drivers, etc) must recover from allocation failure.  That might be much more difficult if taking a lock requires an allocation.

Would it be possible to use an intrusive linked list instead?  This assumes that a mutex cannot be moved while it is locked, which seems like a reasonable assumption ― it will be borrowed by its guard.